src/tools/s2wasm.cpp

```diff
@@ -131,10 +133,17 @@ int main(int argc, const char *argv[]) {
     linker.emscriptenGlue(meta);
   }
 
+  if (options.debug) std::cerr << "Validating..." << std::endl;
+  if (!wasm::WasmValidator().validate(linker.getOutput().wasm)) {
+    std::cerr << "Error: linked module is not valid.\n";
```

Comment:
I did it that way so we would still get the output, for debugging purposes (the exit status is still 1). But since we expect it to be valid now, I could go either way on that.